packages/loader/container-loader/src/container.ts

```diff
@@ -2436,11 +2453,92 @@ export class Container
 	}
 }
 
+/**
+ * Diagnostics information about a given connection instance (single attempt/clientId),
+ * traversing the four connection states (disconnected, establishingConnection, catchingUp, connected).
+ *
+ * Includes what the current connection state is, and what is happening within that state
+ *
+ * @beta
+ */
+export interface ConnectionDiagnostics {
+	/**
+	 * The client id assigned for this connection by the ordering service
+	 * Only set when currentState is "catchingUp" or "connected"
+	 */
+	clientId?: string;
+
+	//* This is hard to compute at the right time.  Maybe can follow example of ReconnectMode below?
+	/** read-only v. read/write connection */
+	connectionMode?: ConnectionMode;
```

Comment:
I'd rename it - it's desired connection mode. What we get depends on permissions and service implementation